src/main/java/io/airlift/slice/Slice.java

```diff
@@ -97,9 +99,21 @@ public static Slice toUnsafeSlice(ByteBuffer byteBuffer)
     private final long retainedSize;
 
     /**
-     * Reference is typically a ByteBuffer object, but can be any object this
-     * slice must hold onto to assure that the underlying memory is not
-     * freed by the garbage collector.
+     * Reference have two use cases:
+     *
+     * 1. It can be an object this slice must hold onto to assure that the
+     * underlying memory is not freed by the garbage collector.
+     * It is typically a ByteBuffer object, but can be any object.
+     *
+     * 2. If reference is not used to prevent garbage collector to free
```

Comment:
Grammar: "If reference is not used to prevent the garbage collector from freeing the underlying memory, ..."